### PR TITLE
make zones work for logged out users

### PR DESCRIPTION
**Pull request by BTBTravis**
Hey @khomtali here is a quick run through to get the zone feature working for logged in and logged out users. This is a pr into your branch. Feel free to approve and merge or just take it as inspiration up to you.

### CODE REVIEW

File: src/components/TrainingPlan/index.js [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
+
 function TrainingPlan() {
+  const { user } = useUserContext();
+
   return (
     <div>
-      <h2>Your training plan</h2>
+      <h2>Your Training Plan</h2>
+      {user ? (
+        <p>feature coming soon!</p>
```
Comment (khomtali) [inline src/components/TrainingPlan/index.js:12]: added this `<p>` in my branch :)

File: src/components/TrainingPlan/index.js [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
+      )
+      :
+      (
+        <p>you must create an account to use this feature</p>
```
Comment (khomtali) [inline src/components/TrainingPlan/index.js:16]: good idea, but actually I didn't want to show all the pages to unauthorised users. let me think about it later

File: src/components/Zones/index.js [hunk 3]
```diff
@@ -10,21 +10,19 @@ import './styles.css';
...
-  // if (user.age && user.resting_heart_rate) {
-  //   const userInput = new ZoneDataModel({ userAge: user.age, restRate: user.resting_heart_rate });
-  //   setZones(userInput.calculateZones);
-  // }
+  useEffect(() => {
+    if (user && user.age && user.resting_heart_rate) {
+      const userInput = new ZoneDataModel({ userAge: user.age, restRate: user.resting_heart_rate });
+      setZones(userInput.calculateZones);
+    }
+  }, [user]);
```
Comment (khomtali) [inline src/components/Zones/index.js:19]: borrowed it! thanks! seems I need to refresh my understanding of `useEffect` 😳

**Comment by BTBTravis** [comment]
Glad it helped!

**Pull request closed without merging**